recipes/kissfft/all/conanfile.py

```diff
@@ -0,0 +1,67 @@
+import os
+from conans import ConanFile, CMake, tools
+
+
+class KissfftConan(ConanFile):
+    name = "kissfft"
+    license = "BSD-3-Clause"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/mborgerding/kissfft"
+    description = "a Fast Fourier Transform (FFT) library that tries to Keep it Simple, Stupid"
+    topics = ("conan",)
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False],
+               "fPIC": [True, False],
+               "datatype": ["float", "double", "int16_t", "int32_t", "SIMD"],
+               "use_alloca": [True, False]}
+    default_options = {"shared": False,
+                       "fPIC": True,
+                       "datatype": "float",
+                       "use_alloca": False}
+    exports_sources = ["CMakeLists.txt"]
+    generators = "cmake",
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "_build_subfolder"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        cmake = CMake(self)
+        cmake.definitions["KISSFFT_PKGCONFIG"] = False
+        cmake.definitions["KISSFFT_STATIC"] = not self.options.shared
+        cmake.definitions["KISSFFT_TEST"] = False
+        cmake.definitions["KISSFFT_TOOLS"] = False
+        cmake.definitions["KISSFFT_DATATYPE"] = self.options.datatype
+        cmake.definitions["KISSFFT_USE_ALLOCA"] = self.options.use_alloca
+        cmake.configure()
```

Comment:
```suggestion
        cmake.configure(build_folder=self._build_subfolder)
```

Reply:
Use what you declared, or remove _build_subfolder: https://github.com/conan-io/conan-center-index/pull/4935/files#diff-15e597ed04b8d70a67350287913ae9236010fd59385df0e59977fc9dc44af661R30

Reply:
nope, build_subfolder doesn't work, already tried it

Reply:
```
CMake Error at CMakeLists.txt:4 (include):
  include could not find load file:

    /Users/jenkins/w/BuildSingleReference@3/.conan/data/kissfft/131.1.0/_/_/build/d8fc853ae566eec1e2198d52080fa29fda09ef83/_build_subfolder/conanbuildinfo.cmake


CMake Error at CMakeLists.txt:5 (conan_basic_setup):
  Unknown CMake command "conan_basic_setup".
```

Reply:
and I don't understand why it should be as you suggest, it works without it, and there is no hook complain about it. I am not sure what should it improve in the first place - only makes code larger.

Reply:
_build_subfolder separates build cache from copied source. No big deal, sometimes is not possible to use _build_subfolder because the project forces CMAKE_SOURCE_DIR/CMAKE_BINARY_DIR.

That's a recommendation, not a rule, that's why is not listed as mandatory on hooks. Also, we only use it for CMake.

Reply:
doesn't seem to be a strong rule:
```
SSE4@DESKTOP-E3NFU1V MINGW64 /c/bincrafters/conan-center-index (theora)
$ grep -RnI "cmake.configure(build_folder=self._build_subfolder)"  . | wc
    233     466   24343

SSE4@DESKTOP-E3NFU1V MINGW64 /c/bincrafters/conan-center-index (theora)
$ grep -RnI "cmake.configure()"  . | wc
    940    1888   72333
```

Reply:
It's a recommendation for better build organization. You know when we are building a package, usually we check build_folder to visually all artifacts generated. If we put all in a separated folder and don't mix with sources would be better to distinguish both. But like you said, it's not a strong rule, it more like a best practice. I'm okay not using it, because it's a feature for package development, not for final user which will install the package.